sklearn/metrics/_classification.py

```diff
@@ -403,7 +415,14 @@ def confusion_matrix(
     prefer_skip_nested_validation=True,
 )
 def multilabel_confusion_matrix(
-    y_true, y_pred, *, sample_weight=None, labels=None, samplewise=False
+    y_true,
+    y_pred,
+    *,
+    sample_weight=None,
+    labels=None,
+    samplewise=False,
+    unique_y_true=None,
+    unique_y_pred=None,
```

Comment:
`labels`, `unique_y_true` and `unique_y_pred` makes the public API looks really bloated. The only alternative I see is to have a private function.

Reply:
I agree with @thomasjpfan. I see only 2 solutions:

- Make some private function where the public one is a wrapper around
- Implementing our own efficient `unique` for NumPy arrays but it could be rather complex